test/common/index.js

```diff
@@ -631,7 +631,7 @@ function expectWarning(name, expected) {
     // Remove a warning message after it is seen so that we guarantee that we
     // get each message only once.
     map.delete(expected);
```

Comment:
This seems to be a bug as well, `expected` is the array of all warnings, and this call returns `false`.